File: dataset_reduzido/avatar/Python/Code/atcoder_ABC146_D.py
from collections import deque
import sys
input = sys.stdin.readline
def bfs ( v , N , G , E ) :
    visited = [ 0 ] * N
    queue = deque ( )
    K = - 1
    node2color = [ - 1 for _ in range ( N ) ]
    queue.append ( v )
    visited [ v ] = 1
    while queue :
        q = queue.popleft ( )
        color = 0
        for nex in G [ q ] :
            if visited [ nex ] :
                continue
            visited [ nex ] = 1
            color += 1
            if color == node2color [ q ] :
                color += 1
            node2color [ nex ] = color
            E [ ( min ( q , nex ) , max ( q , nex ) ) ] = color
            queue.append ( nex )
        K = max ( K , color )
    return K
def main ( ) :
    N = int ( input ( ) )
    G = [ deque ( ) for _ in range ( N ) ]
    E = dict ( )
    for i in range ( N - 1 ) :
        a , b = map ( lambda x : int ( x ) - 1 , input ( ).split ( ) )
        G [ a ].append ( b )
        G [ b ].append ( a )
        E [ ( a , b ) ] = 0
    K = bfs ( 0 , N , G , E )
    print ( K )
    for value in E.values ( ) :
        print ( value )
main ( )

File: output/LLaMa/avatar/Python/Java/atcoder_ABC146_D.java
```java
import java.util.*;
import java.io.*;

public class Main {
    static class FastReader {
        BufferedReader br;
        StringTokenizer st;

        public FastReader() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        String next() {
            while (st == null || !st.hasMoreElements()) {
                try {
                    st = new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() {
            return Integer.parseInt(next());
        }
    }

    static class Pair implements Comparable<Pair> {
        int first;
        int second;

        public Pair(int first, int second) {
            this.first = first;
            this.second = second;
        }

        @Override
        public int compareTo(Pair other) {
            return Integer.compare(this.first, other.first);
        }
    }

    static void bfs(int v, int N, List<List<Integer>> G, Map<Pair, Integer> E) {
        boolean[] visited = new boolean[N];
        Queue<Integer> queue = new LinkedList<>();
        int K = -1;
        int[] node2color = new int[N];
        queue.add(v);
        visited[v] = true;
        while (!queue.isEmpty()) {
            int q = queue.poll();
            int color = 0;
            for (int nex : G.get(q)) {
                if (visited[nex]) {
                    continue;
                }
                visited[nex] = true;
                color++;
                if (color == node2color[q]) {
                    color++;
                }
                node2color[nex] = color;
                E.put(new Pair(Math.min(q, nex), Math.max(q, nex)), color);
                queue.add(nex);
            }
            K = Math.max(K, color);
        }
        return;
    }

    public static void main(String[] args) {
        FastReader in = new FastReader();
        int N = in.nextInt();
        List<List<Integer>> G = new ArrayList<>();
        for (int i = 0; i < N; i++) {
            G.add(new ArrayList<>());
        }
        Map<Pair, Integer> E = new HashMap<>();
        for (int i = 0; i < N - 1; i++) {
            int a = in.nextInt() - 1;
            int b = in.nextInt() - 1;
            G.get(a).add(b);
            G.get(b).add(a);
            E.put